02-peer-protocol.md

```diff
@@ -492,7 +492,10 @@ A sending node:
    (pay to pubkey hash), OR
     2. `OP_HASH160` `20` 20-bytes `OP_EQUAL` (pay to script hash), OR
     3. `OP_0` `20` 20-bytes (version 0 pay to witness pubkey), OR
-    4. `OP_0` `32` 32-bytes (version 0 pay to witness script hash)
+    4. `OP_0` `32` 32-bytes (version 0 pay to witness script hash), OR
+    5. if (and only if) `opt_shutdown_anysegwit` is negotiated:
+      * `OP_1` through `OP_16` inclusive, followed by a single push of 2 to 40 bytes
```

Comment:
Beware that your `is_witness_program()` is bip 141 compliant, otherwise if the witness program size is unbounded the non-funder counterparty might inflate its `scriptpubkey` to burn in fees the funder balance.